web/modules/custom/ingredient_validation/src/Plugin/Validation/Constraint/IngredientConstraintValidator.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ingredient_validation\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the Ingredients field on the Product CT.
+ */
+class IngredientConstraintValidator extends ConstraintValidator {
```

Comment:
In general it's a good practice to make a class final if you don't want other classes to extend this. Can you think of a case when this class would be inherited? It's always easier to 'un-final' a class than have unwanted inheritances.

https://www.php.net/manual/en/language.oop5.final.php